crates/gosub_shared/src/traits/document.rs

```diff
@@ -1,10 +1,7 @@
 use crate::node::NodeId;
 use crate::traits::node::Node;
 
-pub trait DocumentInterface {
+pub trait Document {
     fn get_node_by_id(&self, node_id: NodeId) -> Option<&dyn Node>;
-}
-
-pub trait DocumentHandle {
-    fn get(&self) -> Box<impl DocumentInterface>;
+    fn get_parent_node(&self, node: &dyn Node) -> Option<&dyn Node>;
```

Comment:
You probably can get rid of the `dyn` by letting the document specify its node type like this

```rust
pub trait Document {
    type Node: Node; //the type name is `Node` and after the `:` it says that this type needs to implement the `Node` trait
    fn get_node_by_id(&self, node_id: NodeId) -> Option<&Self::Node>;
    fn get_parent_node(&self, node: &Self::Node) -> Option<&Self::Node>;
}
    
```